src/lib.rs

```diff
@@ -325,8 +313,35 @@ impl BasicDataFormatDescriptor {
             flags: DataFormatFlags::from_bits_truncate(flags),
             texel_block_dimensions,
             bytes_planes,
-            samples,
+            samples_offset: offset,
+            samples_end: initial_offset + descriptor_block_size as usize,
+        }
+    }
+
+    pub fn sample_informations<'data>(&self, data: &'data [u8]) -> impl Iterator<Item = SampleInformation> + 'data {
```

Comment:
Rather than requiring the user to pass in the correct `data`, let's pipe it through `self`. Additionally, "information" is its own plural, so let's drop the trailing s from this function name.

Reply:
> Rather than requiring the user to pass in the correct `data`, let's pipe it through `self`. If the lifetime parameter becomes an issue in practice (which I don't expect) we can split out a separate struct for just the fixed-size parts.

Ok. I was thinking that the lifetime being introduced to the API is not really a problem as you anyway have the data for the file available throughout usage of it. I'll do that.

> Additionally, "information" is its own plural, so let's drop the trailing s from this function name.

Yup.

Reply:
Done.